MovieApp/MovieApp/App/Features/MovieBrowser/Components/SubcategoryView/SubcategoryView+Design.swift

```diff
@@ -0,0 +1,95 @@
+import UIKit
+
+extension SubcategoryView: DesignProtocol {
+    
+    func buildViews() {
+        createViews()
+        styleViews()
+        defineLayoutForViews()
+    }
+    
+    func createViews() {
+        scrollView = UIScrollView()
+        addSubview(scrollView)
+        
+        subcategoriesStack = UIStackView()
+        scrollView.addSubview(subcategoriesStack)
+        
+        addButtons()
+    }
+    
+    func styleViews() {
+        scrollView.contentInset = UIEdgeInsets(top: 0, left: 4 * offset, bottom: 0, right: 0)
+        scrollView.showsHorizontalScrollIndicator = false
+        
+        subcategoriesStack.alignment = .center
+        subcategoriesStack.spacing = 5 * offset
+        subcategoriesStack.axis = .horizontal
+        
+        styleButtons()
+    }
+    
+    func defineLayoutForViews() {
+        scrollView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        subcategoriesStack.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+    }
+    
+}
+
+extension SubcategoryView {
+    
+    func addButtons() {
+        for tag in 0..<maxNumberOfSubcategoryItems {
```

Comment:
`0 ..< maxNumberOfSubcategoryItems`

Reply:
I don't have that [part of the code ](https://github.com/lihter/Movie-App/pull/24/files#diff-b2b46513e62a9e31165f1b658855d74d9d487f3eb70e301c3940cebbcbe2e529L44)anymore. The for loop is now going through all the subcategories in array.